src/storage.h

```diff
@@ -0,0 +1,226 @@
+#ifndef CYCLUS_STORAGES_STORAGE_H_
+#define CYCLUS_STORAGES_STORAGE_H_
+
+#include <string>
+#include <list>
+#include <vector>
+
+#include "cyclus.h"
+
+// forward declaration
+namespace storage {
+class Storage;
+} // namespace storage
+
+
+namespace storage {
+/// @class Storage
+///
+/// This Facility is intended to hold materials for a user specified
+/// amount of time in order to model a storage facility with a certain
+/// residence time or holdup time.
+/// The Storage class inherits from the Facility class and is
+/// dynamically loaded by the Agent class when requested.
+///
+/// @section intro Introduction
+/// This Agent was initially developed to support the fco code-to-code 
+/// comparison.
+/// It's very similar to the "NullFacility" of years 
+/// past. Its purpose is to hold materials and release them only  
+/// after some period of delay time.
+///
+/// @section agentparams Agent Parameters
+/// in_commods is a vector of strings naming the commodities that this facility receives
+/// out_commods is a string naming the commodity that in_commod is stocks into
+/// residence_time is the minimum number of timesteps between receiving and offering
+/// in_recipe (optional) describes the incoming resource by recipe
+/// 
+/// @section optionalparams Optional Parameters
+/// max_inv_size is the maximum capacity of the inventory storage
+/// throughput is the maximum processing capacity per timestep
+///
+/// @section detailed Detailed Behavior
+/// 
+/// Tick:
+/// Nothing really happens on the tick. 
+///
+/// Tock:
+/// On the tock, any material that has been waiting for long enough (delay 
+/// time) is placed in the stocks buffer.
+///
+/// Any brand new inventory that was received in this timestep is placed into 
+/// the processing queue to begin waiting. 
+/// 
+/// Making Requests:
+/// This facility requests all of the in_commod that it can.
+///
+/// Receiving Resources:
+/// Anything of the in_commod that is received by this facility goes into the 
+/// inventory.
+///
+/// Making Offers:
+/// Any stocks material in the stocks buffer is offered to the market.
+///
+/// Sending Resources:
+/// Matched resources are sent immediately.
+class Storage 
+  : public cyclus::Facility,
+    public cyclus::toolkit::CommodityProducer {
+ public:  
+  /// @param ctx the cyclus context for access to simulation-wide parameters
+  Storage(cyclus::Context* ctx);
+  
+  #pragma cyclus decl
+
+  #pragma cyclus note {"doc": "A storage facility converts from one " \
+                              "commodity to another, with an optional delay."}
```

Comment:
It doesn't need to be nearly as long as the reactor's description since it is operationally simpler.